nixos/modules/services/networking/vpp.nix

```diff
@@ -0,0 +1,543 @@
+
+                      {option}`skip-cores` and {option}`workers` are incompatible with {option}`corelist-workers`.
+                    '';
+                  };
+                };
+              };
+          };
+        };
+
+        settingsFile = mkOption {
+          type = types.path;
+          example = "/etc/vpp/startup.conf";
+          default = pkgs.writeText "vpp-${name}.conf" (parseConfig cfg.instances.${name}.settings);
+          defaultText = ''
+            pkgs.writeText "vpp-\${name}.conf" (parseConfig config.services.vpp.instances.\${name}.settings);
+          '';
+          description = ''
+            Configuration file passed to {command}`vpp -c`.
+            It is recommended to use the {option}`settings` option instead.
+
+            Setting this option will override the config file
+            auto-generated from the {option}`settings` option.
+          '';
+        };
+
+        startupConfig = mkOption {
+          type = types.str;
+          default = "";
+          example = ''
+            set interface state TenGigabitEthernet1/0/0 up
+            set interface state TenGigabitEthernet1/0/1 up
+            set interface ip address TenGigabitEthernet1/0/0 2001:db8::1/64
+            set interface ip address TenGigabitEthernet1/0/1 2001:db8:1234::1/64
+          '';
+          description = ''
+            Script to run on startup in {command}`vppctl`, passed to
+            VPP's `startup-config` option in the `unix` section as a file.
+
+            This is used to configure things like IP addresses and routes. To configure
+            interfaces, plugins, etc., see the {option}`settings` option.
+          '';
+        };
+
+        startupConfigFile = mkOption {
+          type = types.path;
+          example = "./vpp-startup.conf";
+          default = pkgs.writeText "vpp-${name}-startup.conf" cfg.instances.${name}.startupConfig;
+          defaultText = ''
+            pkgs.writeText "vpp-\${name}-startup.conf" (builtins.toString config.services.vpp.instances.\${name}.startupConfig);
+          '';
+          description = ''
+            File to run as a script on startup in {command}`vppctl`, passed
+            to VPP's `startup-config` option in the `unix` section.
+
+            Setting this option will override {option}`startupConfig`.
+          '';
+        };
+      };
+    };
+in
+{
+  options.services.vpp = {
+    hugepages = {
+      autoSetup = mkOption {
+        default = false;
+        example = true;
+        description = ''
+          Whether to automatically setup hugepages for use with FD.io's Vector Packet Processor.
+
+          If any programs other than VPP use hugepages or you want to use 1GB hugepages, it's recommended
+          to keep this `false` and set them up manually using {option}`boot.kernel.sysctl` or {option}`boot.kernelParams`.
+        '';
+      };
```

Comment:
For example redis sets this up. Would this work together to make things easy for end users?

Reply:
That I don't know, depends on how exactly VPP and redis use explicit hugepages. The autosetup is only here since VPP *requires* hugepages to even start up properly, and this is significantly easier than having to add all the sysctls for a trivial deployment. More complicated setups like having 2 services that use hugepages are out of scope for this option & indicated as such in the description.